src/signals/settings/users/containers/Overview/index.js

```diff
@@ -8,10 +9,51 @@ import Pager from 'components/Pager';
 
 import PageHeader from 'signals/settings/components/PageHeader';
 import useFetchUsers from './hooks/useFetchUsers';
+import routes from '../../../routes';
 
+const UsersOverview = ({ pageSize, history }) => {
+  const location = useLocation();
+  const { pageNum } = useParams();
+  const [page, setPage] = useState(1);
+  const { isLoading, users } = useFetchUsers({ page, pageSize });
 
-const UsersOverview = ({ page }) => {
-  const { isLoading, users } = useFetchUsers();
+  /**
+   * Get page number value from URL query string
+   *
+   * @returns {number|undefined}
+   */
+  const getPageNumFromQueryString = () => pageNum && parseInt(pageNum, 10);
+
+  // subscribe to 'page' state value changes
+  useEffect(() => {
+    if (history.action === 'POP') {
+      return;
+    }
+
+    const pageNumber = getPageNumFromQueryString();
+
+    if (pageNumber !== page) {
+      history.push(routes.usersPaged.replace(':pageNum', page));
+    }
+  }, [page]);
+
+  // subscribe to 'location' changes
+  useEffect(() => {
+    const pageNumber = getPageNumFromQueryString();
+
+    if (pageNumber !== page) {
+      setPage(pageNumber);
+    }
+  }, [location]);
+
+  const onItemClick = e => {
+    const { dataset } = e.currentTarget;
+    const { itemId } = dataset;
+
+    if (itemId) {
+      history.push(routes.user.replace(':userId', itemId));
+    }
+  };
 
   return (
     <div className="users-overview-page">
```

Comment:
This comment is about the next line
```js
<PageHeader title={`Gebruikers (${users.length})`} />
```

If we don't wait until loading is finished this titel will flash from `0` to actual number

I would also like to suggest showing `30 of 70` instead of just the number of items on the current page. Maybe we need to discuss this with UX